File: Go/Euler006.go
package main

import (
	"fmt"
)

func main() {
	const n = 101
	sumOfSquares := SumOfSquares(1, n)
	squareOfSum := SquareOfSum(1, n)
	result := squareOfSum - sumOfSquares
	fmt.Println(result)
}

func SquareOfSum(start int, end int) int {
	sum := 0
	for i := start; i < end; i++ {
		sum += i
	}
	return sum * sum
}

func SumOfSquares(start int, end int) int {
	sum := 0
	for i := start; i < end; i++ {
		sum += i * i
	}
	return sum
}

File: Kotlin/Euler006.kt
package Euler006

fun main() {
  val n = 100L
  val sumOfSquares = (1L..n).asSequence().map { it -> it * it }.sum()
  val sum = (1L..n).sum()
  val squareOfSum = sum * sum
  println(squareOfSum - sumOfSquares)
}
